crates/client/rpc/src/lib.rs

```diff
@@ -194,6 +199,29 @@ where
             None
         })
     }
+
+    /// Returns the state diff for the given block.
+    ///
+    /// # Arguments
+    ///
+    /// * `block_hash` - The hash of the block containing the state diff (starknet block).
+    fn get_state_diff(&self, starknet_block_hash: &BlockHash) -> Result<StateDiff, String> {
+        let state_diff = self.backend.da().state_diff(starknet_block_hash).unwrap_or_else(|err| {
+            error!("{err}");
```

Comment:
Why are you silencing the error? You can bubble it up. `MissingStateDiff` or something along those lines

Reply:
Let the caller decide if he want to dismiss it or not

Reply:
Fixed